server/controllers/files.ts

```diff
@@ -0,0 +1,81 @@
+import config from 'config';
+import { Request, Response } from 'express';
+
+import { hasUploadedFilePermission } from '../graphql/common/uploaded-file';
+import { idDecode, IDENTIFIER_TYPES } from '../graphql/v2/identifiers';
+import { getSignedGetURL, parseS3Url } from '../lib/awsS3';
+import { UploadedFile } from '../models';
+import { SUPPORTED_FILE_TYPES_IMAGES } from '../models/UploadedFile';
+
+/**
+ * GET /api/files/:uploadedFileId
+ *
+ * Query Params
+ *
+ * json - return json response
+ * thumbnail - return thumbnail json
+ */
+export async function getFile(req: Request, res: Response) {
+  res.set('Cache-Control', 'private');
+
+  const isJsonAccepted = req.query.json !== undefined;
+  const isThumbnail = req.query.thumbnail !== undefined;
+
+  const { uploadedFileId } = req.params;
+  const { expenseId, draftKey } = req.query;
+
+  if (expenseId && typeof expenseId !== 'string') {
+    return res.status(400).send({ message: 'Invalid id' });
+  }
+
+  if (draftKey && typeof draftKey !== 'string') {
+    return res.status(400).send({ message: 'Invalid id' });
+  }
+
+  let decodedId: number;
+  try {
+    decodedId = idDecode(uploadedFileId, IDENTIFIER_TYPES.UPLOADED_FILE);
+  } catch (err) {
+    return res.status(400).send({ message: 'Invalid id' });
+  }
+
+  const uploadedFile = await UploadedFile.findByPk(decodedId);
+
+  if (!uploadedFile) {
+    return res.status(403).send({ message: 'Unauthorized' });
+  }
+
+  const actualUrl = uploadedFile.getDataValue('url');
```

Comment:
We don't want people to start generating signed URLs for public files. If a file is public, we should return early:

```suggestion
  const actualUrl = uploadedFile.getDataValue('url');
  if (uploadedFile.isPublicFile()) {
    return actualUrl;
  }
```

In `server/models/UploadedFile.ts`:
```
  public isPublicFile(): boolean {
    return [
      'ACCOUNT_AVATAR',
      'ACCOUNT_BANNER',
      'ACCOUNT_LONG_DESCRIPTION',
      'TIER_LONG_DESCRIPTION',
    ].includes(this.kind);
  }
```

This also made me realize: https://github.com/opencollective/opencollective/issues/7718.